sonic_installer/bootloader/grub.py

```diff
@@ -85,35 +85,31 @@ def remove_image(self, image):
         run_command('grub-set-default --boot-directory=' + HOST_PATH + ' 0')
         click.echo('Image removed')
 
-    def verify_image_platform(self, image_path):
-        if not os.path.isfile(image_path):
-            return False
-
-        # Get running platform's ASIC
+    def platform_in_platforms_asic(self, platform, image_path):
+        """
+        For those images that don't have devices list builtin, check_output will raise an exception.
+        In this case, we simply return True to make it worked compatible as before.
+        Otherwise, we need to check if platform is inside the supported target platforms list.
+        """
         try:
-            version_info = device_info.get_sonic_version_info()
-            if version_info:
-                asic_type = version_info['asic_type']
-            else:
-                asic_type = None
-        except (KeyError, TypeError) as e:
-            click.echo("Caught an exception: " + str(e))
+            with open(os.devnull, 'w') as fnull:
+                p = subprocess.Popen(["sed", "-e", "1,/^exit_marker$/d", image_path], stdout=subprocess.PIPE, preexec_fn=default_sigpipe)
+                output = subprocess.check_output(["tar", "xf", "-", PLATFORMS_ASIC, -O], stdin=p.stdout, stderr=fnull, preexec_fn=default_sigpipe, text=True)
```

Comment:
>stderr=fnull [](http://example.com/codeflow?start=105&length=12)

Is it the same as the default value `None`?

Reply:
No. I tried locally. `None` still prompt error msg.

Reply:
Then the `with` scope is too small, you need to cover `p2.wait()`, to be safer, cover until function exit